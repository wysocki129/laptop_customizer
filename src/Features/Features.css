.main__form {
	flex: 5;
	padding: 4px;
}

.feature__list {
	list-style: none;
}
