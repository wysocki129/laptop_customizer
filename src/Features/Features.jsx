import React, { Component } from 'react';
import './Features.css';
import Options from '../Options/Options';

class Features extends Component {
	render() {
		const {
			features,
			handleUpdateFeature,
			selectedOptions
		} = this.props;
		const featList = Object.keys(features).map(key => {
			const options = (
				<Options
					selectedOptions={selectedOptions}
					featureType={key}
					allOptions={features[key]}
					handleUpdateFeature={handleUpdateFeature}
				/>
			);
			return (
				<div className="feature" key={key}>
					<div className="feature__name">{key}</div>
					<ul className="feature__list">{options}</ul>
				</div>
			);
		});
		return (
			<section className="main__form">
				<h3>TECH SPECS AND CUSTOMIZATIONS</h3>
				{featList}
			</section>
		);
	}
}

Features.defaultProps = {
	features: {
		Processor: [],
		'Operating Sytstem': [],
		'Video Card': [],
		Display: []
	}
};

export default Features;
