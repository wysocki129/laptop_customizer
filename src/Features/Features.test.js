import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Features from './Features';

const FEATURES = {
	Processor: [
		{
			name:
				'17th Generation Intel Core HB (7 Core with donut spare)',
			cost: 700
		},
		{
			name:
				'Professor X AMD Fire Breather with sidewinder technology',
			cost: 1200
		}
	],
	'Operating System': [
		{
			name: 'Ubuntu Linux 16.04',
			cost: 200
		},
		{
			name: 'Bodhi Linux',
			cost: 300
		}
	],
	'Video Card': [
		{
			name: 'Toyota Corolla 1.5v',
			cost: 1150.98
		},
		{
			name: 'Mind mild breeze 2000',
			cost: 1345
		}
	],
	Display: [
		{
			name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
			cost: 1500
		},
		{
			name: '15.3" HGTV (3840 x 2160) Home makeover edition',
			cost: 1400
		}
	]
};

describe('Features Component', () => {
	it('renders without crashing the app', () => {
		shallow(<Features />);
	});
	it('shows all options for a feature', () => {
		const wrapper = shallow(<Features features={FEATURES} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
