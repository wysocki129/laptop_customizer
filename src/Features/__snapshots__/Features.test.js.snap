// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Component shows all options for a feature 1`] = `
<section
  className="main__form"
>
  <h3>
    TECH SPECS AND CUSTOMIZATIONS
  </h3>
  <div
    className="feature"
    key="Processor"
  >
    <div
      className="feature__name"
    >
      Processor
    </div>
    <ul
      className="feature__list"
    >
      <Options
        allOptions={
          Array [
            Object {
              "cost": 700,
              "name": "17th Generation Intel Core HB (7 Core with donut spare)",
            },
            Object {
              "cost": 1200,
              "name": "Professor X AMD Fire Breather with sidewinder technology",
            },
          ]
        }
        featureType="Processor"
      />
    </ul>
  </div>
  <div
    className="feature"
    key="Operating System"
  >
    <div
      className="feature__name"
    >
      Operating System
    </div>
    <ul
      className="feature__list"
    >
      <Options
        allOptions={
          Array [
            Object {
              "cost": 200,
              "name": "Ubuntu Linux 16.04",
            },
            Object {
              "cost": 300,
              "name": "Bodhi Linux",
            },
          ]
        }
        featureType="Operating System"
      />
    </ul>
  </div>
  <div
    className="feature"
    key="Video Card"
  >
    <div
      className="feature__name"
    >
      Video Card
    </div>
    <ul
      className="feature__list"
    >
      <Options
        allOptions={
          Array [
            Object {
              "cost": 1150.98,
              "name": "Toyota Corolla 1.5v",
            },
            Object {
              "cost": 1345,
              "name": "Mind mild breeze 2000",
            },
          ]
        }
        featureType="Video Card"
      />
    </ul>
  </div>
  <div
    className="feature"
    key="Display"
  >
    <div
      className="feature__name"
    >
      Display
    </div>
    <ul
      className="feature__list"
    >
      <Options
        allOptions={
          Array [
            Object {
              "cost": 1500,
              "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
            },
            Object {
              "cost": 1400,
              "name": "15.3\\" HGTV (3840 x 2160) Home makeover edition",
            },
          ]
        }
        featureType="Display"
      />
    </ul>
  </div>
</section>
`;
